OSRMTextInstructions/OSRMTextInstructions.swift

```diff
@@ -59,10 +100,34 @@ extension String: Tokenized {
         }
         return result
     }
+    
+    func inflected(into variant: String, version: String) -> String {
+        guard let grammar = OSRMTextInstructionsGrammar?[version] as? [String: Any] else {
+            return self
+        }
+        
+        guard let rules = grammar[variant] as? [[String]] else {
+            return self
+        }
+        
+        var grammaticalReplacement = " \(self) "
+        var regularExpressionOptions: NSRegularExpression.Options = []
+        if let meta = OSRMTextInstructionsGrammar?["meta"] as? [String: String],
+            let flags = meta["regExpFlags"] {
+            regularExpressionOptions = NSRegularExpression.Options(javaScriptFlags: flags)
+        }
+        
+        for rule in rules {
+            let regularExpression = try! NSRegularExpression(pattern: rule[0], options: regularExpressionOptions)
+            grammaticalReplacement = regularExpression.stringByReplacingMatches(in: grammaticalReplacement, options: [], range: NSRange(location: 0, length: grammaticalReplacement.characters.count), withTemplate: rule[1])
```

Comment:
Does this method replace all matches not the first only (JS RegExp `g` flag is always on)? I hope this will not cause problems but this should be documented in OSRMTI Grammar feature description.

Reply:
Yes, the `NSRegularExpression.stringByReplacingMatches(in:options:range:withTemplate:)` method always replaces all matches. Replacing just one occurrence would be a more manual affair, either finding the first matching substring and replacing it, or enumerating the matches but stopping the enumeration after the first iteration.

I was thinking that the presence of `^` in each of the regular expression patterns would avoid issues where a rule unexpected matches multiple times, but that would be a brittle assumption unless we remove the `^` anchors from the JSON file and add them to the patterns at runtime.

Reply:
Yes, adding `^` at runtime is bad idea. So we just need to warn developers in Grammar description about implicitly turned on `g` flag in Swift and perhaps add some tests for this